### PR TITLE
MGMT-3356 Cluster CRD - align with K8s API name conventions

### PR DESCRIPTION
**Pull request by nmagnezi**

### CODE REVIEW

File: internal/controller/api/v1alpha1/cluster_types.go [hunk 1]
```diff
@@ -31,18 +31,18 @@ type ClusterSpec struct {
...
-	ApiVip                   string `json:"apiVip,omitempty"`
-	ApiVipDnsName            string `json:"apiVipDnsName,omitempty"`
-	IngressVip               string `json:"ingresVip,omitempty"`
-	MachineNetworkCidr       string `json:"machineNetworkCidr,omitempty"`
-	SshPublicKey             string `json:"sshPublicKey,omitempty"`
-	VipDhcpAllocation        bool   `json:"vipDhcpAllocation,omitempty"`
-	HttpProxy                string `json:"httpProxy,omitempty"`
-	HttpsProxy               string `json:"httpsProxy,omitempty"`
+	ServiceNetworkCIDR       string `json:"serviceNetworkCIDR,omitempty"`
+	APIVIP                   string `json:"APIVIP,omitempty"`
```
Comment (mhrivnak) [inline internal/controller/api/v1alpha1/cluster_types.go:38]: Now we're getting picky... :)

I think the more important aspect of the convention is that the json field name start lower-case. That applies even to acronyms, and even to acronyms followed by other stuff. For example, the Volume resource has a field `awsElasticBlockStore`. Having two acronyms back-to-back is more unusual, but I think the right thing to do would be `apiVIP`.

File: internal/controller/api/v1alpha1/cluster_types.go [hunk 2]
```diff
@@ -31,18 +31,18 @@ type ClusterSpec struct {
...
+	APIVIPDNSName            string `json:"APIVIPDNSName,omitempty"`
+	IngressVIP               string `json:"ingressVIP,omitempty"`
+	MachineNetworkCIDR       string `json:"machineNetworkCIDR,omitempty"`
+	SSHPublicKey             string `json:"SSHPublicKey,omitempty"`
+	VIPDhcpAllocation        bool   `json:"VIPDhcpAllocation,omitempty"`
```
Comment (mhrivnak) [inline internal/controller/api/v1alpha1/cluster_types.go:43]: DHCP needs to be all-caps in both forms.

File: internal/controller/api/v1alpha1/cluster_types.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ type ClusterProgressInfo struct {
 
 type HostNetwork struct {
 	// cidr
-	Cidr string `json:"cidr,omitempty"`
+	CIDR string `json:"CIDR,omitempty"`
```
Comment (mhrivnak) [inline internal/controller/api/v1alpha1/cluster_types.go:63]: For isolated acronyms, the conventions and examples support the json form being all lower-case. Again looking at the Volume resource, it has fields such as `nfs`, `iscsi`, `rbd`, etc.

**Review by mhrivnak — CHANGES_REQUESTED** [review]

File: internal/controller/api/v1alpha1/cluster_types.go [hunk 4]
```diff
@@ -31,18 +31,18 @@ type ClusterSpec struct {
...
-	ApiVipDnsName            string `json:"apiVipDnsName,omitempty"`
-	IngressVip               string `json:"ingresVip,omitempty"`
-	MachineNetworkCidr       string `json:"machineNetworkCidr,omitempty"`
-	SshPublicKey             string `json:"sshPublicKey,omitempty"`
-	VipDhcpAllocation        bool   `json:"vipDhcpAllocation,omitempty"`
-	HttpProxy                string `json:"httpProxy,omitempty"`
-	HttpsProxy               string `json:"httpsProxy,omitempty"`
+	ServiceNetworkCidr       string `json:"serviceNetworkCIDR,omitempty"`
+	APIVip                   string `json:"apiVIP,omitempty"`
+	APIVipDNSName            string `json:"apiVIPdnsName,omitempty"`
```
Comment (nmagnezi) [inline internal/controller/api/v1alpha1/cluster_types.go:39]: @mhrivnak I'm not 100% sure `apiVIPdnsName` is valid, convention-wise. Let me know what you think.

File: internal/controller/api/v1alpha1/cluster_types.go (hunk 1 above)
Comment (nmagnezi) [reply internal/controller/api/v1alpha1/cluster_types.go]: Thanks, Michael. Updated `JSON` names accordingly.

File: internal/controller/api/v1alpha1/cluster_types.go (hunk 2 above)
Comment (nmagnezi) [reply internal/controller/api/v1alpha1/cluster_types.go]: Ack

File: internal/controller/api/v1alpha1/cluster_types.go (hunk 3 above)
Comment (nmagnezi) [reply internal/controller/api/v1alpha1/cluster_types.go]: Ack. that also applies to `cpu` and `mtu` (and more) in the WIP host CRD. will take that into account

**Review by filanov — APPROVED** [review]

**Pull request merged**